File: MSw_implementation/Executive.rb
require_relative 'Board'

#Executive class to run the program
#Will hold a board and handle interactions with front-end

@m_rows = 0
@m_cols = 0
@m_numMines = 0
@gameOver = false
@wonGame = false

#when game starts enter board size and number of mines
def initialize (rows, cols, mines)
    #check if values work
    if rows < 2 || cols < 2
        raise RuntimeError.new("Invalid board size")
    end
    if mines > rows * cols -1 || mines < 1
        raise RuntimeError.new("Invalid quantity of mines")
    end
    
    @m_rows = rows
    @m_cols = cols
    @m_numMines = mines
end

#run the game - CURRENTLY SET UP FOR CONSOLE CONTROL
def run()
    #create board
    b1 = Board.new(@m_rows, @m_cols, @m_numMines)

    #accept starting position from FRONT to initialize mines
    puts "type initial row choice, then col choice"
    rowChoice = gets.chomp
    colChoice = gets.chomp
    b1.placeBombs(rowChoice.to_i, colChoice.to_i)

    #while the game is not over, allow user input
    while !gameOver && !wonGame
        #likely won't need to check validity of input since it's coming from the front end
        puts "enter 1 to toggle a flag or 2 to check for a mine"
        userChoice = gets.chomp.to_i

        if userChoice == 1#user wants to toggle flag
            puts "enter row, then col"
            rowChoice = gets.chomp
            colChoice = gets.chomp
            #toggle the flag for specified space - need func in Board
        elsif userChoice == 2#user checks for mine
            puts "enter row, then col"
            rowChoice = gets.chomp
            colChoice = gets.chomp
        end

        #check every iteration if the game should end

        #update board control file if game does not end
    end

end

File: MSw_implementation/Executive.py
from Board.py import Board #import BoardSpace, getSpace, showSpace

class Executive:

    '''Define member variables'''
    m_rows = 0
    m_cols = 0
    m_numMines = 0
    gameOver = False
    wonGame = False
    m_board = 8
    firstMove = True
    



    def __init__(self, rows, cols, numMines):
        if rows < 2 or cols < 2:
            raise RuntimeError('Invalid board size')
        if numMines < 1 or numMines > rows * cols - 1:
            raise RuntimeError('Invalid number of mines')        

        self.m_rows = rows
        self.m_cols = cols
        self.m_numMines = numMines
        self.gameOver = False
        self.wonGame = False
        self.m_board = Board(self.m_rows,self.m_cols, self.m_numMines)
        self.firstMove = True
        return

    def leftClick(self, row, col):
        '''returns false is user loses game
            returns true if user made a valid move
        '''
        #if this is user's first move, still need to set mines
        if self.firstMove is True:
            firstStep(self.m_board, row, col)
            self.firstMove = False
        else:
            if selectSpace(self.m_Board, row, col) is True:
                return (False)
        return

    def rightClick(self, row, col):
        '''returns -1 if user is out of flags
            returns 0 if flag was successfully planted
            returns 1 if user wins (all mines flagged)
        '''
        try:
            toggleFlagSpace(self.m_board, row, col)
        except:
            return (-1)
        
        #check if user has won the game (flagged all mines)
        if userWin(self.m_board) is True:
            return 1
        else:
            return 0

        



    
